internal/pkg/service/cli/dialog/create_table.go

```diff
@@ -28,49 +40,106 @@ func (p *Dialogs) AskCreateTable(args []string, branchKey keboola.BranchKey, all
 		}
 
 		opts.BucketKey = keboola.BucketKey{BranchID: branchKey.ID, BucketID: bucketID}
-
 		name := p.options.GetString(`name`)
 		if !p.options.IsSet(`name`) {
 			name, _ = p.Ask(&prompt.Question{
 				Label:       "Table name",
 				Description: "Enter the table name.",
+				Validator:   prompt.ValueRequired,
 			})
 		}
 		opts.CreateTableRequest.Name = name
 	}
 
-	columnsStr := p.options.GetString(`columns`)
-	if !p.options.IsSet(`columns`) {
-		columnsStr, _ = p.Ask(&prompt.Question{
-			Label:       "Columns",
-			Description: "Enter a comma-separated list of column names.",
+	// Columns
+	columnsMethod, err := p.columnsDefinitionMethod()
+	if err != nil {
+		return opts, err
+	}
+
+	switch columnsMethod {
+	case columnsNamesFlag:
+		columnsStr := p.options.GetString(`columns`)
+		colNames := strings.Split(strings.TrimSpace(columnsStr), ",")
+		opts.CreateTableRequest.Columns = getOptionCreateRequest(colNames)
+	case columnsDefinitionFlag:
+		filePath := p.options.GetString("columns-from")
+		columnsDefinition, err := parseJSONInputForCreateTable(filePath)
+		if err != nil {
+			return table.Options{}, err
+		}
+		opts.CreateTableRequest.Columns = columnsDefinition
+	case columnsNamesInteractive:
+		columnsStr := p.options.GetString(`columns`)
+		if !p.options.IsSet(`columns`) {
+			columnsStr, _ = p.Ask(&prompt.Question{
+				Label:       "Columns",
+				Description: "Enter a comma-separated list of column names.",
+			})
+		}
+		colNames := strings.Split(strings.TrimSpace(columnsStr), ",")
+		opts.CreateTableRequest.Columns = getOptionCreateRequest(colNames)
+	case columnsDefinitionInteractive:
+		input, _ := p.Editor("yaml", &prompt.Question{
+			Label:       "Columns definition from file",
+			Description: "Columns definition from file",
+			Default:     p.defaultValue(),
```

Comment:
`Columns definition from file` -> `Columns definitions`, the library will add `[press enter to start the editor]`.